### PR TITLE
multiple endpoint support

### LINKED ISSUE #27: Support testing multiple endpoints with the same interface name
Hi,
Since version 1.0.7 support for multiple endpoints with the same interface name has been dropped and we have a couple of use cases for this functionality, for example, [the smtp integrator operator](https://github.com/canonical/smtp-integrator-operator):
```
provides:
  smtp:
    interface: smtp
  smtp-legacy:
    interface: smtp
```
On version 3.2.0 we are seeing the following error, as the framework is not able to determine for which endpoint it need to run the test:
```
ERROR    pytest_interface_tester:plugin.py:332 Interface tester plugin failed with Multiple endpoints found for provider/smtp: ['smtp', 'smtp-legacy']: cannot guess which one it is we're supposed to be testing
Traceback (most recent call last):
  File "/home/arturo/smtp-integrator-operator/.tox/interface/lib/python3.10/site-packages/interface_tester/plugin.py", line 330, in run
    test_fn()
  File "/tmp/tmpj65im5gl/charm-relation-interfaces/interfaces/smtp/v0/interface_tests/test_provider.py", line 18, in test_data_published_on_created
  File "/home/arturo/smtp-integrator-operator/.tox/interface/lib/python3.10/site-packages/interface_tester/interface_test.py", line 267, in run
    state_out = self._run(event)
  File "/home/arturo/smtp-integrator-operator/.tox/interface/lib/python3.10/site-packages/interface_tester/interface_test.py", line 387, in _run
    relations = self._generate_relations_state(
  File "/home/arturo/smtp-integrator-operator/.tox/interface/lib/python3.10/site-packages/interface_tester/interface_test.py", line 484, in _generate_relations_state
    endpoint = self._get_endpoint(supported_endpoints, role, interface_name=interface_name)
  File "/home/arturo/smtp-integrator-operator/.tox/interface/lib/python3.10/site-packages/interface_tester/interface_test.py", line 465, in _get_endpoint
    raise ValueError(
ValueError: Multiple endpoints found for provider/smtp: ['smtp', 'smtp-legacy']: cannot guess which one it is we're supposed to be testing
```

The discussion can be found [here](https://matrix.to/#/!eNCNzcteYUDDYpStsu:ubuntu.com/$b8Rqk4Bd2RSmG1jJTygfQbJZozENZ3NJg_pJp5uTt9A?via=ubuntu.com&via=matrix.org)

### PR DESCRIPTION
**Pull request by PietroPasotti**
Fixes #27 

Kinda hard to test, might want to create a better mock charm.

Manual testing strategy:
check out a charm that has multiple endpoints with the same interface on a single role, then pin this branch in your requirements.txt:

>    "pytest-interface-tester@git+http://github.com/canonical/pytest-interface-tester@multiple-endpoint-support",

### CODE REVIEW

File: interface_tester/interface_test.py [hunk 1]
```diff
@@ -80,6 +80,12 @@ class _InterfaceTestContext:
     state_template: Optional[State]
     """Initial state that this test should be run with, according to the charm."""
 
+    endpoint: str = None
+    """
+    Endpoint being tested. 
+    Only required if there's multiple endpoints with the same interface.
+    """
```
[the remark this answers is not in the document]
Comment (PietroPasotti) [reply interface_tester/interface_test.py]: So I adjusted the mechanism a little bit as it wasn't clear. 
TLDR:
before: we ASSUME there's only one endpoint for each interface/role we want to test, and raise an exception if not
now: by default, we test all endpoints with the given interface/role unless the user has specified 'we only want to test this endpoint'

File: tests/unit/test_e2e.py [hunk 2]
```diff
@@ -523,3 +527,32 @@ def test_data_on_changed():
...
+ from scenario import State, Relation
+
+ from interface_tester.interface_test import Tester
+ from interface_tester.schema_base import DataBagSchema
+
+ def test_data_on_changed():
+     t = Tester(State(
+         relations={{Relation(
+             endpoint='{endpoint}',  # should not matter
+             interface='tracing',
```
[the remark this answers is not in the document]
Comment (PietroPasotti) [reply tests/unit/test_e2e.py]: should be working as intended now
